sphinx/builders/linkcheck.py

```diff
@@ -529,6 +529,15 @@ def run(self, **kwargs: Any) -> None:
                 if uri not in hyperlinks:
                     hyperlinks[uri] = uri_info
 
+        # raw nodes
+        for rawnode in self.document.findall(nodes.raw):
+            uri = rawnode.get('source')
+            if uri:
+                lineno = get_node_line(rawnode)
+                uri_info = Hyperlink(uri, self.env.docname, lineno)
```

Comment:
I believe the hook for `linkcheck-process-uri` should be called with this code path as well.
```python
newuri = self.app.emit_firstresult('linkcheck-process-uri', uri)
```

Reply:
Good catch, as this is done for all types of nodes I took the liberty to refactor this piece of code.

Reply:
That’s always welcome :slightly_smiling_face: